server/src/main/java/org/elasticsearch/action/admin/indices/rollover/RolloverResponse.java

```diff
@@ -19,51 +19,70 @@
 
 package org.elasticsearch.action.admin.indices.rollover;
 
-import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.support.master.ShardsAcknowledgedResponse;
+import org.elasticsearch.common.ParseField;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ObjectParser;
 import org.elasticsearch.common.xcontent.ToXContentObject;
 import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
 
 import java.io.IOException;
-import java.util.AbstractMap;
-import java.util.HashSet;
+import java.util.HashMap;
 import java.util.Map;
 import java.util.Set;
 import java.util.stream.Collectors;
 
-public final class RolloverResponse extends ActionResponse implements ToXContentObject {
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.constructorArg;
 
-    private static final String NEW_INDEX = "new_index";
-    private static final String OLD_INDEX = "old_index";
-    private static final String DRY_RUN = "dry_run";
-    private static final String ROLLED_OVER = "rolled_over";
-    private static final String CONDITIONS = "conditions";
-    private static final String ACKNOWLEDGED = "acknowledged";
-    private static final String SHARDS_ACKED = "shards_acknowledged";
+public final class RolloverResponse extends ShardsAcknowledgedResponse implements ToXContentObject {
+
+    private static final ParseField NEW_INDEX = new ParseField("new_index");
+    private static final ParseField OLD_INDEX = new ParseField("old_index");
+    private static final ParseField DRY_RUN = new ParseField("dry_run");
+    private static final ParseField ROLLED_OVER = new ParseField("rolled_over");
+    private static final ParseField CONDITIONS = new ParseField("conditions");
+
+    @SuppressWarnings("unchecked")
+    private static final ConstructingObjectParser<RolloverResponse, Void> PARSER = new ConstructingObjectParser<>("rollover",
+            true, args -> new RolloverResponse((String) args[0], (String) args[1], (Map<String,Boolean>) args[2],
+            (Boolean)args[3], (Boolean)args[4], (Boolean) args[5], (Boolean) args[6]));
+
+    static {
+        PARSER.declareField(constructorArg(), (parser, context) -> parser.text(), OLD_INDEX, ObjectParser.ValueType.STRING);
+        PARSER.declareField(constructorArg(), (parser, context) -> parser.text(), NEW_INDEX, ObjectParser.ValueType.STRING);
```

Comment:
Can OLD_INDEX or NEW_INDEX also be "null" in any case?

Reply:
no they can't be null